db/migrate/20200511043818_add_more_flair_columns_to_group.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+class AddMoreFlairColumnsToGroup < ActiveRecord::Migration[6.0]
+  def change
+    add_column :groups, :flair_icon, :string
+    add_reference :groups, :flair_image, foreign_key: { to_table: :uploads }
+
+    DB.exec(
+      <<~SQL
+        UPDATE groups SET flair_icon = flair_url
```

Comment:
Since we are migrating the icons to their own column, I think this would be a good time to simplify the format as well. Currently, the accepted formats are "fa-icon" or "fas fa-icon" for solid style icons, "far fa-icon" for regular and "fab fa-icon" for brands. This is finicky (and sadly, it's my fault) and it doesn't work with the icon picker. 

The following one-line change should work: 

```
UPDATE groups SET flair_icon = REPLACE(REPLACE(flair_url, 'fas fa-', ''), ' fa-', '-')
```

After this, the accepted formats are "icon", "far-icon" and "fab-icon". (And the picker enforces the correct format anyway.)